### PR TITLE
ZEPPELIN-3643. Put Shiro related thing in one place

### PR DESCRIPTION
**Pull request by zjffdu**
### What is this PR for?
Move all shiro related thing in one place (`SecurityUtils`)

### What type of PR is it?
[Refactoring]

### Todos
* [ ] - Task

### What is the Jira issue?
* https://issues.apache.org/jira/browse/ZEPPELIN-3643

### How should this be tested?
* CI pass

### Questions:
* Does the licenses files need update? No
* Is there breaking changes for older versions? No
* Does this needs documentation? No

### CODE REVIEW

File: zeppelin-server/src/main/java/org/apache/zeppelin/utils/SecurityUtils.java [hunk 1]
```diff
@@ -179,12 +274,210 @@ public static Collection getRealmsList() {
...
+      }
+
+      if (StringUtils.isBlank(username) || StringUtils.isBlank(tablename)) {
+        return userlist;
+      }
+
+      userquery = String.format("SELECT %s FROM %s", username, tablename);
+    } catch (IllegalAccessException e) {
+      LOGGER.error("Error while accessing dataSource for JDBC Realm", e);
+      return null;
```
Comment (felixcheung) [inline zeppelin-server/src/main/java/org/apache/zeppelin/utils/SecurityUtils.java]: seems like we should return userlist (would be empty) here?
or L475 should also return null?

might just need to be consistent

File: zeppelin-server/src/main/java/org/apache/zeppelin/utils/SecurityUtils.java [hunk 2]
```diff
@@ -179,12 +274,210 @@ public static Collection getRealmsList() {
...
+        if (attrs.get(userAttribute) != null) {
+          String currentUser;
+          if (r.getUserLowerCase()) {
+            LOGGER.debug("userLowerCase true");
+            currentUser = ((String) attrs.get(userAttribute).get()).toLowerCase();
+          } else {
+            LOGGER.debug("userLowerCase false");
+            currentUser = (String) attrs.get(userAttribute).get();
+          }
+          if (LOGGER.isDebugEnabled()) {
```
Comment (felixcheung) [inline zeppelin-server/src/main/java/org/apache/zeppelin/utils/SecurityUtils.java]: btw, why it is calling `isDebugEnabled`?

Comment (zjffdu) [reply zeppelin-server/src/main/java/org/apache/zeppelin/utils/SecurityUtils.java]: I only did refactoring in this PR, didn't change it. It is as before, but I can improve it.

File: zeppelin-server/src/main/java/org/apache/zeppelin/utils/SecurityUtils.java (hunk 1 above)
Comment (zjffdu) [reply zeppelin-server/src/main/java/org/apache/zeppelin/utils/SecurityUtils.java]: will do it

**Pull request closed without merging**